kong/resty/dns/client.lua

```diff
@@ -742,46 +741,38 @@ local function individualQuery(qname, r_opts, try_list)
 end
 
 local queue = setmetatable({}, {__mode = "v"})
--- to be called as a timer-callback, performs a query and returns the results
--- in the `item` table.
-local function executeQuery(premature, item)
-  if premature then return end
 
-  local r, err = resolver:new(config)
-  if not r then
-    item.result, item.err = r, "failed to create a resolver: " .. err
-  else
-    --[[
-    log(DEBUG, PREFIX, "Query executing: ", item.qname, ":", item.r_opts.qtype, " ", fquery(item))
-    --]]
-    add_status_to_try_list(item.try_list, "querying")
-    item.result, item.err = r:query(item.qname, item.r_opts)
-    if item.result then
-      --[[
-      log(DEBUG, PREFIX, "Query answer: ", item.qname, ":", item.r_opts.qtype, " ", fquery(item),
-              " ", frecord(item.result))
-      --]]
-      parseAnswer(item.qname, item.r_opts.qtype, item.result, item.try_list)
-      --[[
-      log(DEBUG, PREFIX, "Query parsed answer: ", item.qname, ":", item.r_opts.qtype, " ", fquery(item),
-              " ", frecord(item.result))
-    else
-      log(DEBUG, PREFIX, "Query error: ", item.qname, ":", item.r_opts.qtype, " err=", tostring(err))
-      --]]
-    end
-  end
+local function queue_get_item(key, qname, r_opts, try_list)
```

Comment:
Move the `timer_at` invocation into this function and rename it `enqueue_query`.  Also rename `item` to `query`.

Reply:
> Move the timer_at invocation into this function 
The timer_at is not the common logic for this function. There are two scenarios of calling this function:
1. queue_get_item -> timer_at(asynchronous query)
2. queue_get_item -> individualQuery (synchronous query)

> rename it `enqueue_query`
No problem, I'll update it later

> Also rename `item` to `query`.

This will introduce many modifications. This pr is an urgent bugfix, we will redesign this module in kong-3.6, I believe it is a good opportunity to refactor this at that time.